test/e2e/framework/util.go

```diff
@@ -2659,51 +2679,37 @@ func ScaleRC(clientset clientset.Interface, internalClientset internalclientset.
 	if !wait {
 		return nil
 	}
-	return WaitForRCPodsRunning(clientset, ns, name)
+	return WaitForControlledPodsRunning(clientset, ns, name, kind)
 }
 
 // Wait up to 10 minutes for pods to become Running.
-func WaitForRCPodsRunning(c clientset.Interface, ns, rcName string) error {
-	rc, err := c.Core().ReplicationControllers(ns).Get(rcName)
+func WaitForControlledPodsRunning(c clientset.Interface, ns, name string, kind schema.GroupKind) error {
+	rtObject, err := getRuntimeObjectForKind(c, kind, ns, name)
 	if err != nil {
+		if apierrs.IsNotFound(err) {
```

Comment:
This means we can't wait for pods, so in my opinion not having this object is an error (even if it doesn't exist). I would remove this if.